src/status_im/contexts/wallet/wallet_connect/core.cljs

```diff
@@ -94,3 +94,14 @@
   (let [test-mode? (get-in db [:profile/profile :test-networks-enabled?])
         networks   (get-in db [:wallet :networks (if test-mode? :test :prod)])]
     (mapv #(-> % :chain-id) networks)))
+
+(defn add-full-testnet-name
+  "Updates the `:full-name` key with the full testnet name if using testnet `:chain-id`.\n
+  e.g. `{:full-name \"Mainnet\"}` -> `{:full-name \"Mainnet Sepolia\"`}`"
+  [network]
+  (let [add-testnet-name (fn [testnet-name]
+                           (update network :full-name #(str % " " testnet-name)))]
+    (condp #(contains? %1 %2) (:chain-id network)
```

Comment:
Hey @clauxx, just food for thought. In my experience with let's say web3 code, we end up with lots and lots of conditionals based on the chain ID. It spreads like a virus, very quickly, because the chain ID is critical to make all sorts of decisions in code. Behavior from test networks often get mixed in with mainnet, it's just not great.

In Clojure we can use protocols, or the more flexible solution with multimethods. Multimethods performance is absolutely fine because we're not making calls in hot and long loops, quite the opposite.

It's worth considering this in future endeavors.

Reply:
Hey! Thanks for the suggestion. I think this might simplify things for sure. I made an [issue](https://github.com/status-im/status-mobile/issues/20775) for this and hopefully we can tackle it sooner than later.

Reply:
Thanks for that @clauxx. When the dust settles this could be interesting. Call me when you think the time is ripe and I can also dive into the code, more like exploratory to see what we can get out of it.